package/MDAnalysis/topology/guessers.py

```diff
@@ -147,13 +147,14 @@ def guess_atom_element(atomname):
         while name:
             if name in tables.elements:
                 return name
-            if name[:-1] in tables.elements:
+            elif name[:-1] in tables.elements:
                 return name[:-1]
-            if name[1:] in tables.elements:
+            elif name[1:] in tables.elements:
                 return name[1:]
-            if len(name) <= 2:
-                return name[0]
-            name = name[:-1]  # probably element is on left not right
```

Comment:
It's a bit silly but you need this for names like "AO5x" which is probably maybe an O (and would return 'AOx' here). But yeah good choice returning an empty string instead of an unphysical element.

Reply:
That makes sense. I was actually wondering why the loop was return that way. What would you suggest I do then? Returning empty string instead of unphysical element would be one of the cases, and another would be return O in cases such as AO5x?